integration-tests/load/automationv2_1/automationv2_1_test.go

```diff
@@ -523,46 +458,55 @@ func TestLogTrigger(t *testing.T) {
 
 	upkeepDelays := make([][]int64, 0)
 	var numberOfEventsEmitted int
-	var batchSize = 500
+	var batchSize uint64 = 500
 
 	for _, gen := range p.Generators {
 		numberOfEventsEmitted += len(gen.GetData().OKData.Data)
 	}
 	numberOfEventsEmitted = numberOfEventsEmitted * numberOfEvents
 	l.Info().Int("Number of Events Emitted", numberOfEventsEmitted).Msg("Number of Events Emitted")
 
-	if endBlock-startBlock < uint64(batchSize) {
-		batchSize = int(endBlock - startBlock)
+	if endBlock-startBlock < batchSize {
+		batchSize = endBlock - startBlock
 	}
 
-	for cIter, consumerContract := range consumerContracts {
+	for _, consumerContract := range consumerContracts {
 		var (
 			logs    []types.Log
 			address = common.HexToAddress(consumerContract.Address())
 			timeout = 5 * time.Second
 		)
-		for fromBlock := startBlock; fromBlock < endBlock; fromBlock += uint64(batchSize) + 1 {
-			var (
-				filterQuery = geth.FilterQuery{
-					Addresses: []common.Address{address},
-					FromBlock: big.NewInt(0).SetUint64(fromBlock),
-					ToBlock:   big.NewInt(0).SetUint64(fromBlock + uint64(batchSize)),
-					Topics:    [][]common.Hash{{consumerABI.Events["PerformingUpkeep"].ID}},
+		for fromBlock := startBlock; fromBlock < endBlock; fromBlock += batchSize + 1 {
+			filterQuery := geth.FilterQuery{
+				Addresses: []common.Address{address},
+				FromBlock: big.NewInt(0).SetUint64(fromBlock),
+				ToBlock:   big.NewInt(0).SetUint64(fromBlock + batchSize),
+				Topics:    [][]common.Hash{{consumerABI.Events["PerformingUpkeep"].ID}},
+			}
+			err = fmt.Errorf("initial error") // to ensure our for loop runs at least once
+			for err != nil {
+				var (
+					logsInBatch []types.Log
+				)
+				ctx2, cancel := context.WithTimeout(ctx, timeout)
```

Comment:
I try to avoid adding numbers to vars like this, is filterCtx okay? Also does the cancel function need to be called immediately or can it be deferred?

Reply:
Copied this part from https://github.com/smartcontractkit/chainlink/blob/develop/integration-tests/testsetups/keeper_benchmark.go#L302-L344 which @kalverra wrote.